src/core_init_atmosphere/mpas_kd_tree.F

```diff
@@ -187,11 +231,25 @@ recursive subroutine mpas_kd_search_internal(kdtree, query, res, distance)
 
       ! Local Values
       real (kind=RKIND) :: current_distance
+      integer :: tie
 
       current_distance = sum((kdtree % point(:) - query(:))**2)
       if (current_distance < distance) then
          distance = current_distance
          res => kdtree
+      else if (current_distance == distance) then
+          !
+          ! Queries that are equal-distant from two points may end up on inteior cells of
+          ! two decompositions. So ties must be broken arbitrarially so different tasks agree
+          ! on the correct location of of a query.
+          !
+          if (associated(res)) then
+              tie = break_tie(res, kdtree)
+              if (tie == 1) then
+                 distance = current_distance
```

Comment:
Is this assignment necessary? L.240 should guarantee that the l.h.s here is the same as the r.h.s.

Reply:
You're correct! Good point!

Reply:
Should we remove this assignment?

Reply:
Yes, looks like I missed it in the last update.